### PR TITLE
Wetland #30 backfill, metals extractor bug fix, lint cleanup, cross-repo ID validator

### PR DESCRIPTION
**Pull request by realmarcin**
## Summary

Four follow-ups against #30 (cross-repo environmental linking) plus a related lint cleanup, bundled because they share test surface.

### 1. #30 Phase 5: wetland backfill (6 communities)

Applies the SPRUCE `related_ingredients` pattern (introduced in #79) to 6 more peatland/wetland communities:

| Community | Ingredients added | Source PMID |
|---|---|---|
| Stordalen Mire Methylotrophic | methanol, methylamines, acetate | PMID:38063415 |
| Prairie Pothole Sulfur/Carbon Virus | sulfate, methanol, ethanol, propan-2-ol | PMID:30086797 |
| MUCC Freshwater Wetland Methane Network | methane, methylated compounds (methanol) | PMID:39843444 |
| Asgard Wetland Soil Methanogenesis-Substrate | acetate, formate, dihydrogen | PMID:39085194 |
| Coastal Forested Wetland Seawater-Ion | sulfate, seawater ions (NaCl), methane | PMID:38628812 |
| Wetland Oxygen-Sulfate GHG | sulfate, dioxygen, lactate, hydrogen sulfide | PMID:38961111 |

All entries use CHEBI terms and snippets taken verbatim from already-cached PubMed abstracts. No `mediaingredientmech_id` values (none have been minted in MIM yet — same dogfood pattern as SPRUCE).

### 2. Metals extractor bug fix + 65-file cleanup

`src/communitymech/metal_extraction.py` used plain substring matching against 2-letter element symbols (`"ti"` for TITANIUM, `"au"` for GOLD). This matched inside unrelated words (`characteristic`, `kinetic`, `australia`, `auto…`), salting `metals_present` with TITANIUM in **56/67** metal-annotated YAMLs and GOLD in several more.

- Fixed: switched to non-alphanumeric-boundary regex matching (case-insensitive). 5 new tests pin the behavior in `tests/test_metal_extraction.py`.
- New `scripts/clean_metals_inplace.py` re-runs extraction and rewrites only the `metals_present` / `rare_earth_elements_present` / `metal_relevance` / `metal_notes` blocks via line-based regex replacement — preserving comments, key order, and unrelated whitespace (unlike `backfill_metals.py`'s `yaml.dump` path).
- Applied across the corpus: **65 community YAMLs** corrected (most lose TITANIUM; a few also lose GOLD; one gains IRON that the old buggy path missed). Diff is uniformly subtractive in the affected lines except for one new tier-1 detection.

### 3. Lint cleanup (`just lint`: ruff/black green)

178 pre-existing ruff errors → 0:

- Removed `T20` (print) from the ruff `select`. Rationale: `src/communitymech/` ships CLI entry points (`cli.py`, `render_*`, `export/*`, `embedding/*`, `validators`) that legitimately use `print` for user output; per-call `# noqa: T201` is louder than the rule is worth.
- Fixed the 44 remaining errors inline: F401 unused rich imports, B904 `raise … from e`, SIM102/SIM103/SIM108 simplifications, C414 redundant `list()` inside `sorted()`, B905 `zip(strict=False)`, E501 line splits (auditor, validators, umap_generator, cli), E402 import order in `batch_reporter.py`.
- Added a per-file E501 ignore for `src/communitymech/llm/prompts.py` (long prompt strings should not be re-wrapped).
- Added targeted `# noqa` with WHY-comments for S301 (pickle from internal cache only), S701 (jinja2 autoescape would break JSON-in-script), S704 (markupsafe.Markup on curator-supplied Mermaid body), S112 (intentional skip-on-parse-fail in id_utils).

`just lint` is still not all-green because **mypy still reports 256 pre-existing errors** (yaml stub missing, implicit Optionals, `Console = None` reassignment). Out of scope for this PR — flagging as separate tech debt.

### 4. Cross-repo ID existence validator (#30 Phase 3, local half)

New module `communitymech.validators.cross_repo_ids` with a two-stage validator:

1. **Pattern check** — `culturemech_id` matches `CultureMech:NNNNNN`, `mediaingredientmech_id` matches `MediaIngredientMech:NNNNNN`. Always runs.
2. **Existence check** — when sibling-repo paths are configured (via flag or `COMMUNITYMECH_SIBLING_REPOS` env var), each ID is looked up in the partner repo's `kb/` dir. Opt-in: with no path configured, the validator emits info-level skip notices rather than silently passing.

CLI: `scripts/validate_cross_repo_ids.py`. Justfile entries: `validate-cross-repo-ids FILE`, `validate-cross-repo-ids-all`. 10 new tests in `tests/test_cross_repo_ids.py` cover patterns, both repos configured, neither configured, malformed IDs, edge cases.

## Test plan

- [x] `just test` — 136 passed, 9 skipped (was 121 → +5 metal_extraction + +10 cross_repo_ids)
- [x] `just validate-all` — all 265 community YAMLs validate against schema
- [x] `just format` — clean
- [x] `uv run ruff check src/ tests/` — clean (was 178 errors → 0)
- [x] `uv run black --check src/ tests/` — clean (49 files)
- [x] `just validate-cross-repo-ids-all` — all clean (no cross-repo IDs to existence-check yet, by design)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: scripts/clean_metals_inplace.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    lines = text.splitlines(keepends=True)
+    new_block = _format_list_block(key, values)
+    return "".join(lines[:start]) + new_block + "".join(lines[end:])
+
+
+def _replace_scalar(text: str, key: str, value: str) -> str:
+    pattern = re.compile(rf"^{re.escape(key)}:.*$", re.MULTILINE)
+    if pattern.search(text):
+        return pattern.sub(f"{key}: {value}", text, count=1)
+    return text + f"{key}: {value}\n"
```
[the remark this answers is not in the document]
Comment (realmarcin) [reply scripts/clean_metals_inplace.py]: Fixed in eaac01a. The rewritten script no longer touches `metal_notes` or `metal_relevance` at all, which sidesteps the multi-line scalar bug entirely — `_replace_scalar` was removed. All previously-corrupted files (Tinto River, Ngawha, Cyprus, Oak Ridge, …) were reverted to main and then re-cleaned with the surgical removal-only logic. Verified: Tinto's `metal_notes` is now a single intact multi-line value, no orphan continuations.

File: scripts/clean_metals_inplace.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    diff_ree = sorted(current_ree) != sorted(ree)
+    if not (diff_metals or diff_ree):
+        return False, ""
+
+    new_text = text
+    new_text = _replace_block(new_text, "metals_present", sorted(metals))
+    new_text = _replace_block(new_text, "rare_earth_elements_present", sorted(ree))
+    new_text = _replace_scalar(new_text, "metal_relevance", relevance)
+    if notes:
+        new_text = _replace_scalar(new_text, "metal_notes", notes)
```
[the remark this answers is not in the document]
Comment (realmarcin) [reply scripts/clean_metals_inplace.py]: Fixed in eaac01a. The rewritten script preserves all curator metadata:

- `metal_notes` and `metal_relevance` are never written. This protects Ngawha's mercury-cycling note, Bayan Obo's REE description, Cyprus's iron-availability prose, etc.
- `metals_present` is only ever shortened, never expanded — and only by removing entries where (a) the extractor keyword list contains a known ambiguous short symbol (`ti`/`au`/`pd`) and (b) the unambiguous keywords (full element name, charged ionic forms) appear nowhere in the file. So Ngawha keeps MERCURY, Oak Ridge keeps URANIUM/IRON/NICKEL/COBALT/ZINC, Bayan Obo keeps its REE list, and so on.
- Adding metals (e.g., the Trichodesmium IRON the fixed extractor would now derive via CHEBI tier-1) is deliberately out of scope; `scripts/backfill_metals.py --dry-run` surfaces those for separate curator review.

Result: 56 files (down from 65 previously), each diff is a 1-2 line removal of TITANIUM/GOLD only. Spot-checked Ngawha, Oak Ridge, SPRUCE, and Bayan Obo — curator content intact.
